wallet/src/actors/worker/methods.rs

```diff
@@ -0,0 +1,263 @@
+    }
+
+    pub fn gen_mnemonic(&self, length: types::MnemonicLength) -> String {
+        let mnemonic = types::MnemonicGen::new().with_len(length).generate();
+        let words = mnemonic.words();
+
+        words.to_string()
+    }
+
+    pub fn flush_db(&self) -> Result<()> {
+        self.wallets.flush_db()?;
+
+        Ok(())
+    }
+
+    pub fn wallet_infos(&self) -> Result<Vec<model::Wallet>> {
+        let wallets = self.wallets.infos()?;
+
+        Ok(wallets)
+    }
+
+    pub fn create_wallet(
+        &mut self,
+        name: Option<String>,
+        caption: Option<String>,
+        password: &[u8],
+        source: &types::SeedSource,
+    ) -> Result<String> {
+        let master_key = crypto::gen_master_key(
+            self.params.seed_password.as_ref(),
+            self.params.master_key_salt.as_ref(),
+            source,
+        )?;
+        let id = crypto::gen_wallet_id(
+            &self.params.id_hash_function,
+            &master_key,
+            self.params.master_key_salt.as_ref(),
+            self.params.id_hash_iterations,
+        );
+
+        let default_account_index = 0;
+        let default_account =
+            account::gen_account(&self.engine, default_account_index, &master_key)?;
+
+        let prefix = id.as_bytes().to_vec();
+        let salt = crypto::salt(&mut self.rng, self.params.db_salt_length);
+        let iv = crypto::salt(&mut self.rng, self.params.db_iv_length);
+        let key = crypto::key_from_password(password, &salt, self.params.db_hash_iterations);
+
+        let wallet_db = db::EncryptedDb::new(self.db.clone(), prefix, key, iv.clone());
+        wallet_db.put("", ())?; // used when unlocking to check if the password is correct
+
+        self.wallets.create(
+            wallet_db,
+            types::CreateWalletData {
+                name,
+                caption,
+                iv,
+                salt,
+                id: &id,
+                account: &default_account,
+            },
+        )?;
+
+        Ok(id)
+    }
+
+    pub fn unlock_wallet(
+        &mut self,
+        wallet_id: &str,
+        password: &[u8],
+    ) -> Result<types::UnlockedSessionWallet> {
+        let (salt, iv) = self
+            .wallets
+            .wallet_salt_and_iv(wallet_id)
+            .map_err(|err| match err {
+                repository::Error::Db(db::Error::DbKeyNotFound) => Error::WalletNotFound,
+                err => Error::Repository(err),
+            })?;
+        let key = crypto::key_from_password(password, &salt, self.params.db_hash_iterations);
+        let session_id = crypto::gen_session_id(
+            &mut self.rng,
+            &self.params.id_hash_function,
+            &key,
+            &salt,
+            self.params.id_hash_iterations,
+        );
+        let prefix = wallet_id.as_bytes().to_vec();
+        let wallet_db = db::EncryptedDb::new(self.db.clone(), prefix, key, iv);
+        wallet_db.get::<_, ()>("").map_err(|err| match err {
```

Comment:
```suggestion
        wallet_db.get::<_, String>(ENCRYPTION_CHECK_KEY).map_err(|err| match err {
```

And verify that the result matches `ENCRYPTION_CHECK_MAGIC`